include/dlaf/matrix/matrix.h

```diff
@@ -384,18 +428,16 @@ Matrix<T, D> createMatrixFromTile(const GlobalElementSize& size, const TileEleme
 ///
 /// @pre @p range must be a valid range for @p matrix
 template <class MatrixLike>
-std::vector<pika::shared_future<typename MatrixLike::ConstTileType>> selectRead(
-    MatrixLike& matrix, common::IterableRange2D<SizeType, LocalTile_TAG> range) {
-  return internal::selectGeneric([&](auto index) { return matrix.read(index); }, range);
+auto selectRead(MatrixLike& matrix, common::IterableRange2D<SizeType, LocalTile_TAG> range) {
+  return internal::selectGeneric([&](auto index) { return matrix.read_sender2(index); }, range);
```

Comment:
Can `class MatrixLike` be replaced by `class T, Device D` and the return type be explicitly specified?

Reply:
you can replace the generic `MatrixLike` with the templated template `MatrixLike<T, D>`.

Reply:
As far as they are not used with `Panel`s...

Reply:
I'm leaving this unchanged in this PR. The comment was just me speculating for myself if it's possible. It is but it seems a bit awkward and I don't want to constrain this to not work with `Panel`s in the future. It's probably no big deal either way, but since it's not required for this PR I'll leave it out.